Result of `nixpkgs-review pr 94113` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
<br>- python27Packages.towncrier
<br>- python37Packages.sunpy
</details>
<details>
  <summary>3 packages built:</summary>
<br>- python37Packages.towncrier
<br>- python38Packages.sunpy
<br>- python38Packages.towncrier
</details>


<pre>
error: build of '/nix/store/zbdwgjx7251z41l9nlpz32br2y7ichld-python3.7-asynctest-0.13.0.drv' on 'ssh://hexa@phoibe.cysec.de' failed: builder for '/nix/store/zbdwgjx7251z41l9nlpz32br2y7ichld-python3.7-asynctest-0.13.0.drv' failed with exit code 1
builder for '/nix/store/zbdwgjx7251z41l9nlpz32br2y7ichld-python3.7-asynctest-0.13.0.drv' failed with exit code 1; last 10 log lines:
      return await coroutine(*args, **kwargs)
    File "/build/asynctest-0.13.0/asynctest/mock.py", line 985, in patched_factory
      patching.__exit__(*exc_info)
    File "/nix/store/wpw6qdjbrv7s72kk81n1lwf3vj6fldk2-python3-3.7.8/lib/python3.7/unittest/mock.py", line 1437, in __exit__
      return exit_stack.__exit__(*exc_info)
    File "/nix/store/wpw6qdjbrv7s72kk81n1lwf3vj6fldk2-python3-3.7.8/lib/python3.7/contextlib.py", line 482, in __exit__
      received_exc = exc_details[0] is not None
  IndexError: tuple index out of range

  ======================================================================
cannot build derivation '/nix/store/gsi09jx8clv9sghab7m94s2shdpv8f5h-python3.7-aioresponses-0.6.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bp3ffc9jgaqsjyjvrarav89112qzx2gd-python3.7-zeep-3.4.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/bxvli87kk2sckdhg7xqcg0dmjzx5vs2g-python2.7-towncrier-19.2.0.drv' on 'ssh://hexa@phoibe.cysec.de' failed: builder for '/nix/store/bxvli87kk2sckdhg7xqcg0dmjzx5vs2g-python2.7-towncrier-19.2.0.drv' failed with exit code 2
builder for '/nix/store/bxvli87kk2sckdhg7xqcg0dmjzx5vs2g-python2.7-towncrier-19.2.0.drv' failed with exit code 1; last 10 log lines:
  /nix/store/11m6l05aqmh81mdxlyjwbq0r4xk1g8xa-python2.7-Twisted-20.3.0/lib/python2.7/site-packages/twisted/trial/unittest.py:12: in <module>
      from twisted.trial._synctest import (
  /nix/store/11m6l05aqmh81mdxlyjwbq0r4xk1g8xa-python2.7-Twisted-20.3.0/lib/python2.7/site-packages/twisted/trial/_synctest.py:17: in <module>
      from twisted.python import failure, log, monkey
  /nix/store/11m6l05aqmh81mdxlyjwbq0r4xk1g8xa-python2.7-Twisted-20.3.0/lib/python2.7/site-packages/twisted/python/log.py:17: in <module>
      from zope.interface import Interface
  E   ImportError: No module named zope.interface
  !!!!!!!!!!!!!!!!!!! Interrupted: 6 errors during collection !!!!!!!!!!!!!!!!!!!!
  =========================== 6 error in 0.59 seconds ============================
  builder for '/nix/store/bxvli87kk2sckdhg7xqcg0dmjzx5vs2g-python2.7-towncrier-19.2.0.drv' failed with exit code 2
</pre>

--- a/pkgs/development/python-modules/sunpy/default.nix
+++ b/pkgs/development/python-modules/sunpy/default.nix
@@ -2,27 +2,29 @@
 , lib
 , buildPythonPackage
 , fetchFromGitHub
-, numpy
-, scipy
-, matplotlib
-, pandas
-, astropy
-, parfive
 , pythonOlder
-, sqlalchemy
-, scikitimage
-, glymur
+
+, asdf
+, astropy
+, astropy-helpers
 , beautifulsoup4
 , drms
-, python-dateutil
-, zeep
-, tqdm
-, asdf
-, astropy-helpers
+, glymur
 , hypothesis
+, matplotlib
+, numpy
+, pandas
+, parfive
 , pytest-astropy
-, pytestcov
 , pytest-mock
+, pytestcov
+, python-dateutil
+, scikitimage
+, scipy
+, sqlalchemy
+, towncrier
+, tqdm
+, zeep
 }:
 
 buildPythonPackage rec {
@@ -47,6 +49,7 @@ buildPythonPackage rec {
     parfive
     sqlalchemy
     scikitimage
+    towncrier
     glymur
     beautifulsoup4
     drms

--- a/pkgs/development/python-modules/towncrier/default.nix
+++ b/pkgs/development/python-modules/towncrier/default.nix
@@ -1,0 +1,40 @@
+{ lib, buildPythonPackage, fetchPypi, isPy27
+, click
+, click-default-group
+, incremental
+, jinja2
+, pytestCheckHook
+, toml
+, twisted
+, git # shells out to git
+}:
+
+buildPythonPackage rec {
+  pname = "towncrier";
+  version = "19.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "15l1gb0hhi9pf3mhhb9vpc93w6w3hrih2ljmzbkgfb3dwqd1l9a8";
+  };
+
+  propagatedBuildInputs = [
+    click
+    click-default-group
+    incremental
+    jinja2
+    toml
+  ];
+
+  # zope.interface collision
+  doCheck = !isPy27;
+  checkInputs = [ git twisted pytestCheckHook ];
+  pythonImportsCheck = [ "towncrier" ];
+
+  meta = with lib; {
+    description = "Utility to produce useful, summarised news files";
+    homepage = "https://github.com/twisted/towncrier/";
+    license = licenses.mit;
+    maintainers = with maintainers; [  ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1559,6 +1559,10 @@ in {
   tokenizers = disabledIf (!isPy3k)
     (toPythonModule (callPackage ../development/python-modules/tokenizers { }));
 
+  towncrier = callPackage ../development/python-modules/towncrier {
+    inherit (pkgs) git;
+  };
+
   transformers = callPackage ../development/python-modules/transformers { };
 
   transforms3d = callPackage ../development/python-modules/transforms3d { };

